system_tests/datastore.py

```diff
@@ -432,6 +432,43 @@ def test_transaction(self):
         self.case_entities_to_delete.append(retrieved_entity)
         self.assertEqual(retrieved_entity, entity)
 
+    def test_transaction_via_explicit_begin_get_commit(self):
```

Comment:
I stripped out the threading complexity from the issue:  I guess I could simplify the testcase method a little further.